tests/integration/buildcmd/build_integ_base.py

```diff
@@ -134,3 +134,20 @@ def _verify_invoke_built_function(self, template_path, function_logical_id, over
 
         process_stdout = process_execute.stdout.decode("utf-8")
         self.assertEqual(json.loads(process_stdout), expected_result)
+
+
+class DedupBuildIntegBase(BuildIntegBase):
```

Comment:
Correct me if I'm wrong, but this does not verify there is no duplicate builds right?

Reply:
This is just BaseClass which contains some shared function definitions there. Actual tests for de-dup builds are here; https://github.com/mndeveci/aws-sam-cli/blob/dedup_builds/tests/integration/buildcmd/test_build_cmd.py#L1037-L1080

Reply:
In those two tests, they just call base _verify_build_and_invoke_functions after running build. How are we checking the intended de-dup feature in which a function should be built only once instead of multiple times?

Reply:
You are right, this is not specifically check for building once, this verifies that building once won't mess-up with different functions. Let me grab command line output and verify there.